### ISSUE TITLE
Brute force protection on login failure

### ISSUE THREAD

**Issue by mafrosis** [mattupstate/flask-security#348]
I don't believe there is anything in Flask-Security that provides IP-based timing delays to prevent brute force login attempts. Please correct me if I'm wrong!

If it's missing, I will implement it in the coming months.

**Comment by nfvs** [comment]
I'm not sure doing this in Flask is a good idea, since you'd be wasting resources that would be best used to process legitimate requests.

This is something that's best done in the web server itself. If you're using Nginx, this can easily be accomplished by using the "ngx_http_limit_req_module" module: http://nginx.org/en/docs/http/ngx_http_limit_req_module.html

**Comment by ldupuy** [comment]
Yes, but a recaptcha after 3 or 5 errors could be great, i think.

**Comment by jonafato** [comment]
I believe this is related to (but not necessarily a duplicate of) #161.

**Comment by cbronazc** [comment]
+1. The nginx or flask-limiter solutions are aimed at rate-limiting a particular IP, but that doesn't address an attacker going after one specific user. I would prefer either to lock the account after *N* failed attempts and force the user to reset via email, or lock account for *N* minutes. Here is a good [example implementation in Ruby.](http://www.rubydoc.info/github/plataformatec/devise/Devise/Models/Lockable)

**Comment by rakeshmukundan** [comment]
I am looking to add "lock" support for a particular user account after N failed login. Anyone else is working on this? if else would love to do it

**Comment by jirikuncar** [comment]
@rakeshmukundan you can try to propose a PR. See my other comment with possible solution using custom login form https://github.com/mattupstate/flask-security/issues/161#issuecomment-310921347. You can also disable account from the form and send them a notification via email.
